main.go

```diff
@@ -127,141 +151,140 @@ func main() {
-		log.Error(ctx.Err(), nil)
+	<-shutdownContext.Done()
+	if shutdownContext.Err() == context.DeadlineExceeded {
+		log.Event(shutdownContext, "shutdown timeout", log.ERROR, log.Error(shutdownContext.Err()))
 	} else {
-		log.Info("done shutdown gracefully", log.Data{"context": ctx.Err()})
+		log.Event(shutdownContext, "done shutdown gracefully", log.ERROR, log.Data{"context": shutdownContext.Err()})
 	}
 	os.Exit(1)
 }
 
-func checkServiceIdentity(ctx context.Context, zebedeeURL, serviceAuthToken string) error {
-	// TODO switch out below to use gedges go-ns package
-	client := rchttp.DefaultClient
+// registerCheckers adds the checkers for the provided clients to the healthcheck object.
+// VaultClient health client will only be registered if encryption is enabled.
+func registerCheckers(ctx context.Context, hc *healthcheck.HealthCheck, isEncryptionEnabled bool,
+	kafkaConsumer *kafka.ConsumerGroup,
+	dimensionExtractedProducer *kafka.Producer,
+	dimensionExtractedErrProducer *kafka.Producer,
+	s3Clients map[string]service.S3Client,
+	vc *vault.Client,
+	zebedeeHealthClient *health.Client,
+	dc *dataset.Client) (err error) {
+
+	if err = hc.AddCheck("Kafka Consumer", kafkaConsumer.Checker); err != nil {
+		log.Event(ctx, "Error Adding Check for Kafka Consumer", log.ERROR, log.Error(err))
+	}
 
-	path := fmt.Sprintf("%s/identity", zebedeeURL)
+	if err = hc.AddCheck("Kafka Producer", dimensionExtractedProducer.Checker); err != nil {
+		log.Event(ctx, "Error Adding Check for Kafka Producer", log.ERROR, log.Error(err))
+	}
 
-	var URL *url.URL
-	URL, err := url.Parse(path)
-	if err != nil {
-		return err
+	if err = hc.AddCheck("Kafka Error Producer", dimensionExtractedErrProducer.Checker); err != nil {
+		log.Event(ctx, "Error Adding Check for Kafka Error Producer", log.ERROR, log.Error(err))
 	}
 
-	req, err := http.NewRequest("GET", URL.String(), nil)
-	if err != nil {
-		return err
+	for bucketName, s3 := range s3Clients {
+		if err = hc.AddCheck(fmt.Sprintf("S3 bucket %s", bucketName), s3.Checker); err != nil {
+			log.Event(ctx, "Error Adding Check for S3 Client", log.ERROR, log.Error(err))
+		}
 	}
-	req.Header.Set(authorizationHeader, serviceAuthToken)
 
-	res, err := client.Do(ctx, req)
-	if err != nil {
-		return err
+	if isEncryptionEnabled {
+		if err = hc.AddCheck("Vault", vc.Checker); err != nil {
+			log.Event(ctx, "Error Adding Check for Vault", log.ERROR, log.Error(err))
+		}
+	}
+
+	if err = hc.AddCheck("Zebedee", zebedeeHealthClient.Checker); err != nil {
+		log.Event(ctx, "Error Adding Check for Zebedee", log.ERROR, log.Error(err))
 	}
-	defer res.Body.Close()
 
-	if res.StatusCode != http.StatusOK {
-		return fmt.Errorf("invalid status [%d] returned from [%s]", res.StatusCode, zebedeeURL)
+	if err = hc.AddCheck("Dataset API", dc.Checker); err != nil {
+		log.Event(ctx, "Error Adding Check for Dataset API", log.ERROR, log.Error(err))
 	}
 
-	log.Info("dimension extractor has a valid service account", nil)
-	return nil
+	return
```

Comment:
Won't this function only return an error if the last call to `hc.AddCheck` returns an error?

Put another way - if the first add fails, but the next succeeds then the err variable will be nil?

Reply:
This is true for all the services that have similar functions!
I will fix this, and then the other PRs - we might want to check other services as well

Reply:
Oh dear, good spot Carl - yes we will need to highlight this as tech debt to go clean up other repos